### PR TITLE
Port of fork

### PR DESCRIPTION
**Pull request by mikebell90**
I've moved this to an official Opentable repo.

Our "base" branch is ForkBase, and I've changed the settings thereof.
The first merge is all the changes to make it 3.9.9 (compatible)

### CODE REVIEW

**Comment by mikebell90** [comment]
@dkaukov To remove confusion, unless you are using opentable/metrics-spring, can you please archive that repo?

File: README.md [hunk 1]
```diff
@@ -19,13 +19,13 @@ This module does the following:
 
 ### Maven
 
-Current version is 3.1.3, which is compatible with Metrics 3.1.2
+Current version is 3.9.9, which is compatible with Metrics 4.0.2
```
Comment (adamzr) [inline README.md:22]: Just go to 4.0.0

File: pom.xml [hunk 2]
```diff
@@ -27,10 +27,10 @@
 		<maven>3.0.0</maven>
 	</prerequisites>
 	<scm>
-		<connection>scm:git:git://github.com/ryantenney/metrics-spring.git</connection>
-		<developerConnection>scm:git:git@github.com:ryantenney/metrics-spring.git</developerConnection>
+		<connection>scm:git:git://github.com/opentable/metrics-spring-1.git</connection>
```
Comment (adamzr) [inline pom.xml:30]: Why `-1`?

File: src/test/java/com/ryantenney/metrics/spring/ReporterTest.java [hunk 3]
```diff
@@ -120,7 +120,7 @@ public void realReporters() throws Throwable {
 			Assert.assertNotNull(ctx.getBean(CsvReporter.class));
 			Assert.assertNotNull(ctx.getBean(JmxReporter.class));
 			Assert.assertNotNull(ctx.getBean(Slf4jReporter.class));
-			Assert.assertNotNull(ctx.getBean(GangliaReporter.class));
+			//Assert.assertNotNull(ctx.getBean(GangliaReporter.class));
```
Comment (adamzr) [inline src/test/java/com/ryantenney/metrics/spring/ReporterTest.java:123]: Delete it!

**Review by adamzr — APPROVED** [review]
Looks good. I wonder if we should consider this a permanent replacement to metrics spring. In that case we should take steps like: 
 - Rename packages to our own name. This will force people to update their code, but it should be a simple update. This would reduce confusion.
 - Rename artifact to our own name.
 - Remove XML support
 - Switch to OTJ parent
 - Open source the project

**Comment by mikebell90** [comment]
@adamzr That all fits my thoughts exactly, though it's damn tricky to prevent them from using the "wrong" metrics-spring, so I was putting off repackaging. Agreed that it looks to me like we could delete like 65% of these stuff (xml support, non graphite [it's all outdated and we don't have time to support). Somewhat ambivalent about otj-parent but not against.

**Pull request merged**